force-app/main/default/classes/WeeklyIterator_TEST.cls

```diff
@@ -0,0 +1,101 @@
+@IsTest
+public with sharing class WeeklyIterator_TEST {
+    @IsTest
+    private static void shouldAssignRule() {
+        RecurrenceRule rRule = new RecurrenceRule();
+        WeeklyIterator iterator = new WeeklyIterator(rRule);
+        System.assertEquals(rRule, iterator.rRule, 'Should assign rule on construction.');
+    }
+
+    /* FUNCTIONAL TESTS */
+
+    @IsTest
+    private static void shouldIterateEveryOtherMonday5TimesOverDaylightSavings() {
+        // Start on Sept 1 2020, 2PM
+        DateTime dateTimeStart = DateTime.newInstance(2020, 9, 1, 14, 0, 0);
+        // every other Monday, 5 times or Oct 8 2099
+        RecurrenceRule rRule = new RecurrenceRule()
+            .withRuleString(
+                'FREQ=WEEKLY;COUNT=5;INTERVAL=2;BYDAY=MO;UNTIL=20991008T100000Z'
+            );
+        List<DateTime> expected = new List<DateTime>{
+            DateTime.newInstance(2020, 9, 14, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 28, 14, 0, 0),
+            DateTime.newInstance(2020, 10, 12, 14, 0, 0),
+            DateTime.newInstance(2020, 10, 26, 14, 0, 0),
+            DateTime.newInstance(2020, 11, 9, 14, 0, 0) // crosses DST
+        };
+        WeeklyIterator iterator = new WeeklyIterator(rRule);
+
+        Test.startTest();
+        List<DateTime> actual = iterator.iterate(dateTimeStart);
+        Test.stopTest();
+
+        System.assertEquals(
+            expected,
+            actual,
+            'List of Dates returned from the iterator was not as expected.'
+        );
+    }
+
+    @IsTest
+    private static void shouldIterateEveryMondayAndTuesdayUntilGivenDate() {
+        // Start on Sept 1 2020, 2PM
+        DateTime dateTimeStart = DateTime.newInstance(2020, 9, 1, 14, 0, 0);
+        // every Monday and Tuesday until October 8 or 30 times
```

Comment:
Can we have a similar test but instead of repeating Monday, repeat Tuesday and Sunday? Just to test the start of week is on Monday type deal when we pass in a Sunday.